src/view/to-do/to-do.view.tsx

```diff
@@ -0,0 +1,84 @@
+import React, { useCallback, useState } from "react";
+import { ToDoHeader } from "../../components/to-do-header/to-do-header.component";
+import { ToDoList } from "../../components/to-do-list/to-do-list.component";
+import { ItemListModel } from "../../model/item-list.model";
+import { v4 as uuidv4 } from 'uuid';
+import { ItemListActions } from "../../components/to-do-list/to-do-item-list/to-do-item-list.component";
+import { ThreeWaySwitch } from "../../components/three-way-switch/three-way-switch.component";
+import { ThreeWaySwitchContainer, ToDoCotainer, ToDoCotainerShadow } from "./to-do.styles";
+
+export type ThreeWayToggleType = 'all' | 'done' | 'not done';
+
+export const ToDoView: React.FC = () => {
+
+    const [array_list, setArrayList] = useState<ItemListModel[]>([]);
+    const [three_way_toggle, setThreeWayToggle] = useState<ThreeWayToggleType>('all');
+
+    const handleNewItem = (item_value: string) => {
+        if (item_value.trim() !== '') {
+            setArrayList([new ItemListModel(uuidv4(), item_value, false), ...array_list]);
+        }
+    }
+
+    const handleItemActions = (item: ItemListModel, action: ItemListActions) => {
+        let new_array = [...array_list];
+        switch (action) {
+            case 'Update':
+                new_array = new_array.map(obj => obj.id === item.id ? item : obj);
+                setArrayList(new_array);
+                break;
+            case 'Delete':
+                setArrayList(new_array.filter(obj => obj.id !== item.id));
+                break;
+            case 'Done':
+                new_array = new_array.map(obj => obj.id === item.id ? { ...obj, done: !obj.done } : obj);
+                setArrayList(new_array);
+                break;
+            default:
+                break;
+        }
+    }
+
+    const handleToggleChange = (val: ThreeWayToggleType) => {
+        setThreeWayToggle(val);
+    }
+
+    const handleToggledArrayList = useCallback(
```

Comment:
I don't think useCallback is needed here. The dependencies are the only 2 state variables in this component, and the callback will be re-defined every time one of the 2 state variables changes. The component doesn't have any props, so that means the callback will be redefined at every render.
Also, if anything I would maybe use useMemo instead of useCallback, because you are not sending this function as a callback, but you are calculating the return value and then sending it into ToDoList.

Reply:
Right, I recognise my mistake. The callback was not needed. I'm sorry I didn't realise it before.